### PR TITLE
StatusButton component

### PR DESCRIPTION
**Pull request by ritch**
## Release Notes

- **New Features**
	- Introduced a new `StatusButton` component for rendering buttons with status indicators.
	- Added `StatusButtonView` component to encapsulate button logic based on schema properties.
	- Enhanced theme with a new `primaryMedium` color for better customization.

- **Improvements**
	- Simplified import paths for `StatusButton` and `StatusButtonView` components.
	- Updated logic to handle custom themes in the `ThemeProvider`.

- **Bug Fixes**
	- Removed unused `Loading` component import in `Embeddings`.

These updates enhance user interaction and improve the overall functionality of the application.

### CODE REVIEW

File: app/packages/components/src/components/StatusButton/StatusButton.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+};
+
+type Severity = "info" | "success" | "warning" | "error";
+
+const SEVERITY_COLORS = {
+  info: "info.main",
+  success: "success.main",
+  warning: "warning.main",
+  error: "error.main",
+  disabled: "text.disabled",
```
Comment (minhtuev) [inline app/packages/components/src/components/StatusButton/StatusButton.tsx:30]: nice!

File: app/packages/components/src/components/StatusButton/StatusButton.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+const ButtonStylesOverrides: ButtonProps["sx"] = {
+  color: (theme) => theme.palette.text.secondary,
+  border: "1px solid",
+  borderColor: (theme) => `${theme.palette.divider} !important`,
+  textTransform: "none",
+  fontSize: "1rem",
+};
+
+type Severity = "info" | "success" | "warning" | "error";
```
Comment (minhtuev) [inline app/packages/components/src/components/StatusButton/StatusButton.tsx:23]: probably need to add `disabled` and `enabled` here too

Comment (ritch) [reply app/packages/components/src/components/StatusButton/StatusButton.tsx:23]: yep! thanks

**Review by minhtuev — APPROVED** [review]
Pulled the change and verified that the button works, thanks Ritchie! ✅ 

Will leave the FE review to @imanjra and @CamronStaley 👍

**Pull request merged**